src/mpl/configure.ac

```diff
@@ -1080,13 +1080,17 @@ if test "$have_gpu" = "no" ; then
     PAC_PUSH_FLAG(CPPFLAGS)
     PAC_APPEND_FLAG([-D__HIP_PLATFORM_AMD__], [CPPFLAGS])
     PAC_CHECK_HEADER_LIB_OPTIONAL([hip],[hip/hip_runtime_api.h],[amdhip64],[hipStreamSynchronize])
-    PAC_POP_FLAG(CPPFLAGS)
+
+    if test "X${pac_have_hip}" != "Xyes" ; then
+        # recover the CPPFLAGS
+        PAC_POP_FLAG(CPPFLAGS)
+    fi
```

Comment:
Having `PUSH` with unbalanced `POP` is just a weird pattern. Why don't you leave the "POP" here, and add back the define in the following -- if that is your intention.

Reply:
fixed this. I realized `__HIP_PLATFORM_AMD__` is still needed for build MPICH. But I did it with `-D` option so that it is explicit to user.